components/nimbus/src/stateful/nimbus_client.rs

```diff
@@ -161,7 +158,25 @@ impl NimbusClient {
     ) -> Result<()> {
         self.read_or_create_nimbus_id(db, writer, state)?;
         self.update_ta_install_dates(db, writer, state)?;
-        self.event_store.lock().unwrap().read_from_db(db)?;
+        self.event_store
+            .lock()
+            .expect("unable to lock event_store mutex")
+            .read_from_db(db)?;
+
+        if let Some(recorded_context) = &self.recorded_context {
+            let targeting_helper = self.create_targeting_helper_with_context(serde_json::to_value(
+                state.targeting_attributes.clone(),
+            )?);
```

Comment:
You don't need to clone, since there is a blanket impl for [`&T where T: Serialize`](https://docs.rs/serde/1.0.209/serde/ser/trait.Serialize.html#impl-Serialize-for-%26T)

```suggestion
            let targeting_helper = self.create_targeting_helper_with_context(serde_json::to_value(
                &state.targeting_attributest,
            )?);
```

Reply:
It can't move `state`, so I do have to

Reply:
You don't have to move state, `&state.targeting_attributes` borrows.